homeassistant/components/hydrawise.py

```diff
@@ -0,0 +1,173 @@
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ACCESS_TOKEN): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+            cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Hunter Hydrawise component."""
+    conf = config[DOMAIN]
+    access_token = conf.get(CONF_ACCESS_TOKEN)
+    scan_interval = conf.get(CONF_SCAN_INTERVAL)
+
+    try:
+        from hydrawiser.core import Hydrawiser
+
+        hydrawise = Hydrawiser(user_token=access_token)
+        hass.data[DATA_HYDRAWISE] = HydrawiseHub(hydrawise)
+    except (ConnectTimeout, HTTPError) as ex:
+        _LOGGER.error(
+            "Unable to connect to Hydrawise cloud service: %s", str(ex))
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+        return False
+
+    def hub_refresh(event_time):
+        """Call Hydrawise hub to refresh information."""
+        _LOGGER.debug("Updating Hydrawise Hub component")
+        hass.data[DATA_HYDRAWISE].data.update_controller_info()
+        dispatcher_send(hass, SIGNAL_UPDATE_HYDRAWISE)
+
+    # Call the Hydrawise API to refresh updates
+    track_time_interval(hass, hub_refresh, scan_interval)
+
+    return True
+
+
+class HydrawiseHub(object):
+    """Representation of a base Hydrawise device."""
+
+    def __init__(self, data):
+        """Initialize the entity."""
+        self.data = data
+
+
+class HydrawiseEntity(Entity):
+    """Entity class for Hydrawise devices."""
+
+    def __init__(self, data, sensor_type):
+        """Initialize the Hydrawise entity."""
+        self.data = data
+        self._sensor_type = sensor_type
+        self._name = "{0} {1}".format(
+            self.data.get('name'), KEY_MAP.get(self._sensor_type))
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_HYDRAWISE, self._update_callback)
+
+    def _update_callback(self):
+        """Call update method."""
+        self.schedule_update_ha_state(True)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the units of measurement."""
+        return UNIT_OF_MEASUREMENT_MAP.get(self._sensor_type)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        temp = self.hass.data['hydrawise'].data.\
+            controller_info['controllers'][0]['last_contact_readable']
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            'identifier': self.data.get('relay'),
+            'last contact:': temp
```

Comment:
Looks like that this should be `'last_contact'`.

Reply:
It was intended to be that way. I changed the ' to " to make it clearer.